[BUG] Does not work with anyio 4

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq.html#frequently-asked-questions) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Schemathesis does not start when anyio 4 is installed.
It seems there is a dependency on `starlette-testclient` which is the cause of the problem.

### To Reproduce

```
from schemathesis import from_asgi, experimental

from starlette.applications import Starlette

schema = from_asgi("/openapi.json", Starlette())


@schema.parametrize()
def test_api(case):
    response = case.call_asgi()
    case.validate_response(response)
```

### Expected behavior

Works

### Environment

```
- OS: [e.g. Linux or Windows]
- Python version: [e.g. 3.11]
- Schemathesis version: [e.g. 3.20.0]
- Spec version: [e.g. Open API 3.0.2]
```

### Additional context

```
../../../../../Library/Caches/pypoetry/virtualenvs/bla-Ms7_xa4C-py3.9/lib/python3.9/site-packages/starlette_testclient/_testclient.py:450: in _portal_factory
    with anyio.start_blocking_portal(**self.async_backend) as portal:
E   AttributeError: module 'anyio' has no attribute 'start_blocking_portal'
```

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -8,6 +8,7 @@ Changelog
 
 - ``GenerationConfig.headers.strategy`` attribute for customizing header generation. :issue:`2137`
 - Support for ``python -m schemathesis.cli``. :issue:`2142`
+- Support for ``anyio>=4.0``. :issue:`2081`
 
 .. _v3.27.0:
 

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -31,7 +31,6 @@ classifiers = [
     "Topic :: Software Development :: Testing",
 ]
 dependencies = [
-    "anyio<4",
     "backoff>=2.1.2,<3.0",
     "click>=7.0,<9.0",
     "colorama>=0.4,<1.0",
@@ -48,7 +47,7 @@ dependencies = [
     "PyYAML>=5.1,<7.0",
     "requests>=2.22,<3",
     "starlette>=0.13,<1",
-    "starlette-testclient<1",
+    "starlette-testclient>=0.4.0,<1",
     "tomli-w>=1.0.0,<2.0",
     "tomli>=2.0.1,<3.0",
     "werkzeug>=0.16.0,<4",

